src/client/src/orangehrmCorporateDirectoryPlugin/pages/CorporateDirectory.vue

```diff
@@ -0,0 +1,312 @@
+    'employee-details': EmployeeDetails,
+    'oxd-loading-spinner': Spinner,
+  },
+
+  props: {
+    jobTitles: {
+      type: Array,
+      default: () => [],
+    },
+    locations: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  setup() {
+    const {$t} = usei18n();
+    const employeeDataNormalizer = data => {
+      return data.map(item => {
+        return {
+          id: item.empNumber,
+          employeeName:
+            `${item.firstName} ${item.middleName} ${item.lastName} ` +
+            (item.terminationId ? $t('general.past_employee') : ''),
+          employeeJobTitle: item.jobTitle?.title,
+          employeeSubUnit: item.subunit?.name,
+          employeeLocation: item.location?.name,
+        };
+      });
+    };
+    const filters = ref({...defaultFilters});
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/directory/employees',
+    );
+    const limit = 8; // this is a static limit since no pagination
+    const state = reactive({
+      employees: [],
+      total: 0,
+      colSize: 4,
+      windowWidth: 0,
+      currentIndex: -1,
+      isLoading: false,
+      offset: 0,
+    });
+
+    const fetchData = () => {
+      state.isLoading = true;
+      http
+        .getAll({
+          limit: limit,
+          offset: state.offset,
+          locationId: filters.value.locationId?.id,
+          empNumber: filters.value.employeeNumber?.id,
+          jobTitleId: filters.value.jobTitleId?.id,
+        })
+        .then(response => {
+          const {data, meta} = response.data;
+          state.total = meta?.total || 0;
+          if (Array.isArray(data)) {
+            state.employees = [
+              ...state.employees,
+              ...employeeDataNormalizer(data),
+            ];
+          }
+        })
+        .finally(() => (state.isLoading = false));
+    };
+
+    const {scrollerRef} = useInfiniteScroll(() => {
+      if (state.employees.length >= state.total) return;
+      state.offset += limit;
+      fetchData();
+    });
+
+    return {
+      scrollerRef,
+      ...toRefs(state),
+      fetchData,
+      filters,
+    };
+  },
+
+  computed: {
+    isMobile() {
+      return this.windowWidth < 600;
+    },
+    hasCurrentIndex() {
+      return this.currentIndex >= 0;
+    },
+  },
+
+  beforeMount() {
+    this.fetchData();
+
+    this.onResize();
+    window.addEventListener('resize', this.onResize);
+  },
+  beforeUnmount() {
+    window.removeEventListener('resize', this.onResize);
+  },
+  methods: {
+    hideEmployeeDetails() {
+      this.currentIndex = -1;
+      this.colSize = 4;
+    },
+    showEmployeeDetails(index) {
+      if (this.currentIndex != index) {
+        this.currentIndex = index;
+        this.colSize = 3;
+      } else {
+        this.currentIndex = -1;
+        this.hideEmployeeDetails();
+      }
+    },
+    onResize() {
+      this.windowWidth = window.innerWidth;
+    },
```

Comment:
*
https://github.com/orangehrm/orangehrm/pull/1295/files#diff-a1e43950d810209d761b5f824c0c8e7dbf3699469a22cb5193d88735b743bfefR245-R260
Ignore this comment